RecoTracker/DeDx/plugins/DeDxHitInfoProducer.cc

```diff
@@ -70,54 +68,34 @@ void DeDxHitInfoProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSe
-  if(useTrajectory)iEvent.getByToken(m_trajTrackAssociationTag, trajTrackAssociationHandle);
-
   // creates the output collection
   auto resultdedxHitColl = std::make_unique<reco::DeDxHitInfoCollection>();
 
   std::vector<int> indices;
 
-  TrajTrackAssociationCollection::const_iterator cit;
-  if(useTrajectory)cit = trajTrackAssociationHandle->begin();
   for(unsigned int j=0;j<trackCollection.size();j++){            
      const reco::Track& track = trackCollection[j];
 
      //track selection
      if(track.pt()<minTrackPt ||  std::abs(track.eta())>maxTrackEta ||track.numberOfValidHits()<minTrackHits){
-        cit++;
         indices.push_back(-1);
         continue;
      }
 
      reco::DeDxHitInfo hitDeDxInfo;
- 
-     if(useTrajectory){  //trajectory allows to take into account the local direction of the particle on the module sensor --> muc much better 'dx' measurement
-        const edm::Ref<std::vector<Trajectory> > traj = cit->key; cit++;
-        const vector<TrajectoryMeasurement> & measurements = traj->measurements();
-        for(vector<TrajectoryMeasurement>::const_iterator it = measurements.begin(); it!=measurements.end(); it++){
-           const TrajectoryStateOnSurface& trajState=it->updatedState();
-           if( !trajState.isValid()) continue;
-     
-           const TrackingRecHit* recHit=(*it->recHit()).hit();
-           if(!recHit)continue;
-           const LocalVector& trackDirection = trajState.localDirection();
-           float cosine = trackDirection.z()/trackDirection.mag();           
-
-           processHit(recHit, trajState.localMomentum().mag(), cosine, hitDeDxInfo, trajState.localPosition());
-        }
-
-     }else{ //assume that the particles trajectory is a straight line originating from the center of the detector  (can be improved)
+     auto const & trajParams = track.extra()->trajParams();
+     // here till debug finishes
+     if (trajParams.size()!=track.recHitsSize()) std::cout << "oops " << iEvent.getProvenance(trackCollectionHandle.id()) << std::endl;
+     assert(trajParams.size()==track.recHitsSize());
+     auto hb = track.recHitsBegin();
         for(unsigned int h=0;h<track.recHitsSize();h++){
-           const TrackingRecHit* recHit = &(*(track.recHit(h)));
-           auto const & thit = static_cast<BaseTrackerRecHit const&>(*recHit);
-           if(!thit.isValid())continue;//make sure it's a tracker hit
+           auto recHit = *(hb+h);
+           if(!recHit->isValid()) continue;
 
-           const GlobalVector& ModuleNormal = recHit->detUnit()->surface().normalVector();         
-           float cosine = (track.px()*ModuleNormal.x()+track.py()*ModuleNormal.y()+track.pz()*ModuleNormal.z())/track.p();
+           auto trackDirection = trajParams[h].direction();
+           float cosine = trackDirection.z()/trackDirection.mag();
  
            processHit(recHit, track.p(), cosine, hitDeDxInfo, LocalPoint(0.0,0.0));
```

Comment:
since we fall on this branch instead of the removed ```if(useTrajectory){ ```
we now get pos at (0,0,0).
Is this variable needed ? 
@quertenmont 

If needed, we'd need to recover it before moving on with this PR.